telegram/ext/callbackcontext.py

```diff
@@ -73,13 +73,35 @@ def __init__(self, dispatcher):
             raise ValueError('CallbackContext should not be used with a non context aware '
                              'dispatcher!')
         self._dispatcher = dispatcher
-        self.chat_data = None
-        self.user_data = None
+        self._chat_data = None
+        self._user_data = None
         self.args = None
         self.matches = None
         self.error = None
         self.job = None
 
+    @property
+    def chat_data(self):
+        return self._chat_data
+
+    @chat_data.setter
+    def chat_data(self, value):
+        if self._chat_data is not None:
```

Comment:
I think this should just always warn. There is no scenario in which setting chat_data is usefull right?

Reply:
There is. It is called every time there is an chat and/or user. I dont understand it either, but this check is needed and it works :D

Reply:
https://github.com/python-telegram-bot/python-telegram-bot/pull/1495/files#diff-a47242a7a011f5b57df445ace75583baR119
It's because of these lines. Makes these set `self._chat/user_data` and you can remove this check

Reply:
you just dont like my solution :(

yeah, yours is better, bah. *grummel*